### PR TITLE
Fix signing tool

### PR DESCRIPTION
**Pull request by Tbaut**
fix https://github.com/polkadot-js/apps/issues/1139
added a help label for account.

Question before I implement it:
Rather than letting users click on the "Unlock" button, I'd fire up directly the unlock modal if a locked account is selected. If it's cancelled, then the unlock button will still be there. I would add a message also to say that the data can't be signed unless the account is unlocked (again it would only be visible if the modal is cancelled). What do you think?

### CODE REVIEW

**Comment by jacogr** [comment]
Sounds fine. I think it is a bit weird to select and have a modal pop-up without your explicit consent, but let's play and see.

File: packages/app-toolbox/src/Sign.tsx [hunk 1]
```diff
@@ -61,15 +57,16 @@ class Sign extends React.PureComponent<Props, State> {
...
     const { t } = this.props;
-    const { defaultValue } = this.state;
+    const { currentPair } = this.state;
 
     return (
       <div className='ui--row'>
         <InputAddress
           className='full'
-          defaultValue={defaultValue}
+          defaultValue={currentPair}
```
Comment (jacogr) [inline packages/app-toolbox/src/Sign.tsx]: Not sure how this works - InputAddress takes a string or UInt8Array.

**Comment by Tbaut** [comment]
Agreed although if users select an account it's most probably because they want to sign something with it, by poping up the modal it saves a click and some reading that then need to unlock the account. We'll see how unexpected this modal ends up being.

**Comment by Tbaut** [comment]
I am surprised that [`onChangeAddress`](https://github.com/polkadot-js/apps/blob/tbaut-fix-siging/packages/app-toolbox/src/Sign.tsx#L204) is called when rendering the page, without any user interaction.

From my understanding It is supposed to get called when the input field value changes, so when `currentPair` changes. The later doesn't change according to my logs... So why does `onChangeAddress` get called?

**Comment by jacogr** [comment]
It actually gets called consistently with all inputs - for address it means that it will initialize with the last selected value. Since the components are typically used uncontrolled (ie. No value prop) this makes sense, otherwise the user will either start with an empty selection or (just as bad) start from the top again.

**Comment by jacogr** [comment]
I think the problem, not selection is because we are passing thought he default. It will be on the last account, so actually no need to pass it at all.

**Comment by Tbaut** [comment]
Thanks, the input is actually behaving as expected now (as weird as it can seem it also did with my 1st commit).

I want to toggle the unlock modal when a locked account is selected but only if the users manually select one, not if the default account is locked (when the page is loaded). In that case they would see a warning and could click on the `Unlock` button if needed. To implement this I wanted to set a state variable at `onChangeAddress` to toggle the unlock modal only if users do change the address, but that's simply not possible this way.

**Comment by Tbaut** [comment]
>so actually no need to pass it at all.

Indeed, it also works well this way.

**Comment by Tbaut** [comment]
What do you think of that? It's in my last commit, if you want to play with. Probably needs some color tweaks:

image

**Comment by jacogr** [comment]
That actually works. Nicely descriptive, still allows selection.

File: packages/app-toolbox/src/Sign.tsx [hunk 2]
```diff
@@ -48,36 +73,41 @@ class Sign extends React.PureComponent<Props, State> {
   }
 
   render () {
+    const { isLocked } = this.state;
+
     return (
       <div className='toolbox--Sign'>
         {this.renderAccount()}
+        <Wrapper>
         {this.renderInput()}
```
Comment (jacogr) [inline packages/app-toolbox/src/Sign.tsx]: Indent this and the next line.

File: packages/app-toolbox/src/Sign.tsx [hunk 3]
```diff
@@ -2,27 +2,55 @@
...
+
+  .unlock-overlay {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    top:0;
+    left:0;
+    background-color: #0f0e0e7a;
+  }
+  .unlock-overlay-warning {
```
Comment (jacogr) [inline packages/app-toolbox/src/Sign.tsx]: (nit) Extra blank line before (and before the next class following)

File: packages/app-toolbox/src/Sign.tsx [hunk 4]
```diff
@@ -86,13 +116,18 @@ class Sign extends React.PureComponent<Props, State> {
...
-      <Button.Group>
-        <Button
-          isPrimary
-          onClick={this.toggleUnlock}
-          label={t('Unlock account')}
-        />
-      </Button.Group>
+      <div className='unlock-overlay-warning'>
+        <div className='unlock-overlay-content'>
+          You need to unlock this account to be able to sign data.<br/>
```
Comment (jacogr) [inline packages/app-toolbox/src/Sign.tsx]: i18n text via `t`

**Pull request merged**